database/migrations/sql/20180603_create_initial_schema.sql

```diff
@@ -0,0 +1,461 @@
+  modified_at     TIMESTAMP WITH TIME ZONE,
+  deleted_at      TIMESTAMP WITH TIME ZONE,
+
+  CONSTRAINT user_account_access_role_to_user_fk         FOREIGN KEY (user_account_id) REFERENCES user_account ON DELETE CASCADE,
+  CONSTRAINT user_account_access_role_to_access_role_fk  FOREIGN KEY (access_role_id)  REFERENCES access_role  ON DELETE CASCADE,
+  CONSTRAINT user_account_access_role_to_organisation_fk FOREIGN KEY (organisation_id) REFERENCES organisation ON DELETE CASCADE
+);
+CREATE TRIGGER update_user_account_access_role_modified_at BEFORE UPDATE ON user_account_access_role
+  FOR EACH ROW EXECUTE PROCEDURE update_modified_at_column();
+
+
+CREATE TABLE permission (
+  permission_id SERIAL NOT NULL UNIQUE,
+  entity        VARCHAR NOT NULL UNIQUE,
+  read_access   permission_type NOT NULL,
+  write_access  permission_type NOT NULL,
+  delete_access permission_type NOT NULL,
+
+  CONSTRAINT permission_pk PRIMARY KEY (permission_id)
+);
+
+
+CREATE TABLE access_role_permission (
+  access_role_id INT NOT NULL,
+  permission_id  INT NOT NULL,
+
+  CONSTRAINT access_role_permission_to_access_role_fk FOREIGN KEY (access_role_id) REFERENCES access_role ON DELETE CASCADE,
+  CONSTRAINT access_role_permission_to_permission_fk  FOREIGN KEY (permission_id)  REFERENCES permission  ON DELETE CASCADE
+);
+
+
+CREATE TABLE user_secret_reset (
+  user_secret_reset_id SERIAL NOT NULL UNIQUE,
+  user_account_id      INT NOT NULL,
+  reset_token          VARCHAR NOT NULL,
+  created_at           TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT now(),
+  modified_at          TIMESTAMP WITH TIME ZONE,
+  deleted_at           TIMESTAMP WITH TIME ZONE,
+
+  CONSTRAINT user_secret_reset_pk                 PRIMARY KEY (user_secret_reset_id),
+  CONSTRAINT user_secret_reset_to_user_account_fk FOREIGN KEY (user_account_id) REFERENCES user_account ON DELETE CASCADE
+);
+CREATE TRIGGER update_user_secret_reset_modified_at BEFORE UPDATE ON user_secret_reset
+  FOR EACH ROW EXECUTE PROCEDURE update_modified_at_column();
+
+
+CREATE TABLE user_invitation (
+  user_invitation_id SERIAL NOT NULL UNIQUE,
+  invitation_status  invitation_statuses NOT NULL,
+  created_at         TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT now(),
+  modified_at        TIMESTAMP WITH TIME ZONE,
+  deleted_at         TIMESTAMP WITH TIME ZONE,
+
+  CONSTRAINT user_invitation_pk PRIMARY KEY (user_invitation_id)
+);
+CREATE TRIGGER update_user_invitation_modified_at BEFORE UPDATE ON user_invitation
+  FOR EACH ROW EXECUTE PROCEDURE update_modified_at_column();
+
+
+CREATE TABLE login_event (
+  login_event_id  SERIAL NOT NULL UNIQUE,
+  user_account_id INT NOT NULL,
+  created_at      TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT now(),
+  deleted_at      TIMESTAMP WITH TIME ZONE,
+
+  CONSTRAINT login_event_pk         PRIMARY KEY (login_event_id),
+  CONSTRAINT login_event_to_user_fk FOREIGN KEY (user_account_id) REFERENCES user_account ON DELETE CASCADE
```

Comment:
Do we want on delete cascade? All user data wants to be soft deleted #5, we could either remove the users row but keep their associated data, or keep their row but overwrite their personal information (name, email, number).

Reply:
Prefer the latter approach. If we don't cascade delete we'll either have unfulfilled constraints which will throw errors, or we'll have ex-foreign keys in tables which relate to a ghost row, which seems quite annoying if we ever want to pull that data together to use.